csharp/src/Google.Protobuf/FieldCodec.cs

```diff
@@ -182,7 +182,7 @@ public static FieldCodec<ulong> ForUInt64(uint tag)
         /// <returns>A codec for the given tag.</returns>
         public static FieldCodec<float> ForFloat(uint tag)
         {
-            return new FieldCodec<float>(input => input.ReadFloat(), (output, value) => output.WriteFloat(value), CodedOutputStream.ComputeFloatSize, tag);
+            return new FieldCodec<float>(input => input.ReadFloat(), (output, value) => output.WriteFloat(value), CodedOutputStream.ComputeFixedFloatSize(0f), tag);
```

Comment:
If you know the size of the float will be 4, you should just write 4 like in ForFixed32 and ForSFixed32 (this applies to ForDouble and ForBool as well).

Reply:
I was thinking about this. I suppose I was struggling with  why the compute methods should take a value, if they do nothing with it. I assumed the design was to ensure  others didn't write code calling the wrong method, compiler error. So then I thought about hard coding the size here, and my thought is, then if someone changes the size of an encoded value down the road inside ComputeSize, they need to also be aware the value is hard coded in FieldCodec and make the change. As I've written it, a change to CodedOutputStream could be done without touching the FieldCodec logic. I don't like passing in the dummy value either, ideally the fixed size methods wouldn't take an argument. 

I will change this to be hard coded as you've asked.

Reply:
The "compute size" methods are very old and outdated. Back in [jskeet/protobuf-csharp-port](https://github.com/jskeet/protobuf-csharp-port/blob/master/src/ProtocolBuffers/CodedOutputStream.ComputeSize.cs) they included the size of a tag with the result of the calculation (as the summary implies). But the methods have been changed since then and no longer reflect that design. So it doesn't make sense to call them if they return a fixed value.

Reply:
ok sounds good